mautic/mautic
# Fix DNC save entity

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
While we use https://github.com/mautic/mautic/pull/7567 in production we noticed something wrong with add DNC to contact. I noticed some issues and suggest improvements. 

If contact has already bounced, bounced flag was removed but and unsubscribe flag not added. In fact, this contact begin contactable

- This PR fixed it persits = true
- This PR rework campaign DNC actions for persist = false


[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Create contact and add bounce flag to contact (If you are not dev,  maybe by sending email to not exists email)
2. Open contact profile and Contact preferences 

![image](https://user-images.githubusercontent.com/462477/74467486-70f09800-4e99-11ea-8853-32b0e098ae03.png)

3. Just save it
4. See Bounced flag removed from contact profile
5. Same happen If on public preferences center and also if you're using campaign action (https://github.com/mautic/mautic/pull/7567 )


#### Steps to test PR:
1. Repeat all steps like for tests
2. After save preferences center (public, in Mautic) you should still see bounced flag in contact profile. After remove bounced flag, then you can add DNC flag by contact preferences center
3. You can also test campaign actions https://github.com/mautic/mautic/pull/7567
- Add Do Not Contact - should add unsubscribe flag - If contact already has bounced flag, no unsubscribe flag added
- Remove Do Not Contact - should remove just Do Not Contact flag from contact profile, not bounce flag
